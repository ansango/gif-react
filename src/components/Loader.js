export const TitleLoader = () => {
  return (
    <div className="border border-green-300 shadow rounded-md p-4 w-full mx-auto">
      <div className="animate-pulse flex space-x-4">
        <div className="flex-1 space-y-4 py-1">
          <div className="space-y-2">
            <div className="h-4 bg-green-400 rounded"></div>
            <div className="h-4 bg-green-400 rounded w-5/6"></div>
          </div>
        </div>
      </div>
    </div>
  );
};

export const CardLoader = () => (
  <div className="border border-green-300 shadow rounded-md p-4 w-full mx-auto">
    <div className="animate-pulse flex space-x-4">
      <div className="flex-1 space-y-4 py-1">
        <div className="h-4 bg-green-400 rounded w-3/4"></div>
        <div className="space-y-2">
          <div className="h-4 bg-green-400 rounded"></div>
          <div className="h-4 bg-green-400 rounded w-5/6"></div>
        </div>
      </div>
    </div>
  </div>
);

const Loader = ({ amount = 1, Component }) => {
  return (
    <>
      {Array(amount)
        .fill()
        .map((_, index) => {
          return <Component key={index} />;
        })}
    </>
  );
};
export default Loader;
