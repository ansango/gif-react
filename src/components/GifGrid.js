import useFetchGifs from "../hooks/useFetchGifs";
import { ContainerGrid, ContainerTitle } from "./Container";
import GifGridItem from "./GifGridItem";
import Loader, { CardLoader, TitleLoader } from "./Loader";

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      {loading && (
        <>
          <ContainerTitle>
            <Loader Component={TitleLoader} />
          </ContainerTitle>
          <ContainerGrid>
            <Loader amount={12} Component={CardLoader} />
          </ContainerGrid>
        </>
      )}
      {!loading && images.length > 0 && (
        <>
          <ContainerTitle>
            <h2 className="font-bold text-2xl py-5">{category}</h2>
          </ContainerTitle>
          <ContainerGrid>
            {images.map((img) => (
              <GifGridItem key={img.id} {...img} />
            ))}
          </ContainerGrid>
        </>
      )}
      {!loading && images.length === 0 && (
        <ContainerTitle>
          <h2 className="font-bold text-2xl py-5">{category} not found</h2>
        </ContainerTitle>
      )}
    </>
  );
};
export default GifGrid;
