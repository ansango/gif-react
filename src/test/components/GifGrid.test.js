import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import Loader from "../../components/Loader";

import useFetchGifs from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("Test <GifGrid/>", () => {
  const category = "Simpsons";

  test("should be render", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show items when useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/cualquier/cosa.jpg",
        title: "Cualquier cosa",
      },
      {
        id: "123",
        url: "https://localhost/cualquier/cosa.jpg",
        title: "Cualquier cosa",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper.find(Loader).exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });

  test("should show Loader when useFetchGifs", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper.find(Loader).exists()).toBe(true);
  });
});
