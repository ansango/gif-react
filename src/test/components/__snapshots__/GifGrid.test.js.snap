// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid/> should be render 1`] = `
<Fragment>
  <ContainerTitle>
    <Loader
      Component={[Function]}
    />
  </ContainerTitle>
  <ContainerGrid>
    <Loader
      Component={[Function]}
      amount={12}
    />
  </ContainerGrid>
</Fragment>
`;
